subsys/bluetooth/services/fast_pair/fp_auth.c

```diff
@@ -85,11 +87,11 @@ BT_CONN_CB_DEFINE(conn_callbacks) = {
 enum bt_security_err auth_pairing_accept(struct bt_conn *conn,
 					 const struct bt_conn_pairing_feat *const feat)
 {
-	/* While Fast Pair authentication module is handling the Bluetooth authentication callbacks,
-	 * peers that do not follow Fast Pair procedure should be rejected. Otherwise invalid
-	 * capabilities will be reported by the Fast Pair Provider.
-	 */
-	return is_conn_handled(conn) ? BT_SECURITY_ERR_SUCCESS : BT_SECURITY_ERR_UNSPECIFIED;
+	if (is_conn_handled(conn) && (feat->io_capability != BT_SMP_IO_NO_INPUT_OUTPUT)) {
```

Comment:
Don't we need to receive a certain value of IO capabilities to guarantee compatibility between them and local value?
Or do we always declare the BT_SMP_IO_KEYBOARD_DISPLAY in our response?

Reply:
The spec explicitly states that we should reject peer if it reports no input no output IO capabilities, so I would leave it as it is here. Provider's IO capabilities rely on set of used authentication callbacks, so there is no need for additional checks here.

The MITM must be required, so I added select to ensure user would not accidentally disable this in Kconfig.